src/main/java/org/folio/service/transactions/batch/BatchEncumbranceService.java

```diff
@@ -0,0 +1,113 @@
+package org.folio.service.transactions.batch;
+
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.Encumbrance;
+import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.model.Transaction.TransactionType;
+
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import java.util.List;
+import java.util.Map;
+
+import static org.folio.rest.jaxrs.model.Transaction.TransactionType.ENCUMBRANCE;
+import static org.folio.utils.CalculationUtils.calculateBudgetSummaryFields;
+import static org.folio.utils.MoneyUtils.subtractMoney;
+import static org.folio.utils.MoneyUtils.sumMoney;
+
+public class BatchEncumbranceService extends AbstractBatchTransactionService {
+
+  @Override
+  public TransactionType getTransactionType() {
+    return ENCUMBRANCE;
+  }
+
+  @Override
+  public void updatesForCreatingTransactions(List<Transaction> transactionsToCreate, BatchTransactionHolder holder) {
```

Comment:
this method name does not sound like a verb, propose to rename to:

- makeUpdatesForCreatingTransactions()
- makeUpdatesForUpdatingTransactions()

Reply:
How about "prepareCreatingTransactions()" ?